### PR TITLE
User account page now dispatches the close action

### LINKED ISSUE #4545: Closing the user account "checkout" modal fails
When we display the CC form, the button does not "close" the modal.
We should make sure it actually closes (by either showing no other modal, or going back to the paywall modal).

image

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR actually hooks up `mapDispatchToProps` in the account iframe, so now clicking the `X` will close the modal.

For the moment, I'm only allowing the close button _after_ the user has logged in, because there won't be a way to bring it back up without it. A future PR will account for that (probably in tandem with #4452 and #4202 ).

# Issues

Fixes #4545 

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/__tests__/components/content/AccountContent.test.tsx [hunk 1]
```diff
@@ -144,4 +145,40 @@ describe('AccountContent', () => {
...
+    })
+
+    it('should setStripe and clearInterval when the window does contain stripe', () => {
+      expect.assertions(2)
+
+      jest.useFakeTimers()
+      const interval = setInterval(() => {}, 15000)
+      const setStripe = jest.fn()
+      const Stripe = jest.fn()
+      ;(Stripe as any).version = 16000
```
Comment (julien51) [inline unlock-app/src/__tests__/components/content/AccountContent.test.tsx]: Meh. this syntax is weird :/

File: unlock-app/src/components/content/AccountContent.tsx [hunk 2]
```diff
@@ -141,7 +150,26 @@ export const mapDispatchToProps = (dispatch: any) => ({
   dismissPurchaseModal: () => dispatch(dismissPurchaseModal()),
 })
 
-export default withConfig(connect(mapStateToProps)(AccountContent))
+export const getStripeHelper = (
```
Comment (julien51) [inline unlock-app/src/components/content/AccountContent.tsx]: Would it be worth moving to its own file?

File: unlock-app/src/__tests__/components/content/AccountContent.test.tsx (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/__tests__/components/content/AccountContent.test.tsx]: 🤔 we could start using semicolons at the end of every line

File: unlock-app/src/components/content/AccountContent.tsx (hunk 2 above)
Comment (cnasc) [reply unlock-app/src/components/content/AccountContent.tsx]: At the moment this is a small enough thing that I think it's ok to leave it here. If we ever start using Stripe across more than one page, then I think it's definitely worth pulling out.

**Pull request merged**